AsyncDisplayKit/ASTextNode.mm

```diff
@@ -749,32 +785,28 @@ - (NSArray *)_rectsForTextRange:(NSRange)textRange measureOption:(ASTextKitRende
   return adjustedRects;
 }
 
-- (NSArray *)rectsForTextRange:(NSRange)textRange
-{
-  return [self _rectsForTextRange:textRange measureOption:ASTextKitRendererMeasureOptionCapHeight];
-}
-
-- (NSArray *)highlightRectsForTextRange:(NSRange)textRange
-{
-  return [self _rectsForTextRange:textRange measureOption:ASTextKitRendererMeasureOptionBlock];
-}
-
 - (CGRect)trailingRect
 {
+  std::lock_guard<std::recursive_mutex> l(_textLock);
+  
   CGRect rect = [[self _renderer] trailingRect];
-  return [self.class _adjustRendererRect:rect forShadowPadding:self.shadowPadding];
+  return ASTextNodeAdjustRenderRectForShadowPadding(rect, self.shadowPadding);
 }
 
 - (CGRect)frameForTextRange:(NSRange)textRange
 {
+  std::lock_guard<std::recursive_mutex> l(_textLock);
```

Comment:
We need the locks  as `ASTextNodeRenderer` is not thread safe at the moment. This was a conscious decision to not make `ASTextNodeRenderer` and `ASTextKitContext` (except the access to the text kit components) thread safe for now and provide the locking in the `ASTextNode` for both components.